pkg/streamingpromql/functions/common.go

```diff
@@ -43,3 +46,7 @@ func FloatTransformationDropHistogramsFunc(transform func(f float64) float64) In
 func Passthrough(seriesData types.InstantVectorSeriesData, _ *pooling.LimitingPool) (types.InstantVectorSeriesData, error) {
 	return seriesData, nil
 }
+
+// RangeVectorStepFunction is a function to operate on a range vector step.
+// floatBuffer and histogramBuffer will contain the values for the step range.
```

Comment:
This line (`floatBuffer and histogramBuffer ...`) is a bit misleading: it might make it sound like implementations of this method don't need to consider `step.RangeEnd`.